drivers/power/ltc4296/ltc4296.c

```diff
@@ -0,0 +1,1099 @@
+	ret = ltc4296_reg_read(dev, port_addr, &val16);
+	if (ret)
+		return ret;
+
+	if ((val16 & LTC4296_NEW_MSK) == LTC4296_NEW_MSK)
+		/* A new ADC value is available */
+		*port_i_out_ma =( ((val16 & 0x0FFF) - LTC4296_ADC_OFFSET) *
+				  ltc4296_spoe_rsense[port_no] /
+				  10);
+	else
+		return ADI_LTC_INVALID_ADC_PORT_CURRENT;
+
+	return 0;
+}
+
+/**
+ * @brief Stimulate valid wake-up signature for a port
+ * @param dev - The device structure
+ * @param port_no - The port number
+ * @param mode - The device configuration mode
+ * @return 0 in case of success, negative code otherwise
+ */
+int ltc4296_port_prebias(struct ltc4296_dev *dev, enum ltc4296_port port_no,
+			 enum ltc4296_config mode)
+{
+	int ret;
+	uint8_t port_addr = 0;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ltc4296_get_port_addr(port_no, LTC_PORT_CFG1, &port_addr);
+	if (ret)
+		return ret;
+
+	if (mode == LTC_CFG_SCCP_MODE)
+		return ltc4296_reg_write(dev, port_addr, 0x0108);
+	else if (mode == LTC_CFG_APL_MODE)
+		return ltc4296_reg_write(dev, port_addr, 0x0109);
+	else
+		return -EINVAL;
+}
+
+/**
+ * @brief Enable specific port
+ * @param dev - The device structure
+ * @param port_no - The port number
+ * @return 0 in case of success, negative code otherwise
+ */
+int ltc4296_port_en(struct ltc4296_dev *dev, enum ltc4296_port port_no)
+{
+	int ret;
+	uint8_t port_addr = 0;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ltc4296_get_port_addr(port_no, LTC_PORT_CFG0, &port_addr);
+	if (ret)
+		return ret;
+
+	return ltc4296_reg_write(dev, port_addr, (LTC4296_SW_EN_MSK |
+				 LTC4296_SW_POWER_AVAILABLE_MSK |
+				 LTC4296_SW_PSE_READY_MSK |
+				 LTC4296_TMFVDO_TIMER_DISABLE_MSK));
+}
+
+/**
+ * @brief Enable specific port and classify
+ * @param dev - The device structure
+ * @param port_no - The port number
+ * @return 0 in case of success, negative code otherwise
+ */
+int ltc4296_port_en_and_classification(struct ltc4296_dev *dev,
+				       enum ltc4296_port port_no)
+{
+	int ret;
+	uint8_t port_addr = 0;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ltc4296_get_port_addr(port_no, LTC_PORT_CFG0, &port_addr);
+	if (ret)
+		return ret;
+
+	return ltc4296_reg_write(dev, port_addr,
+				 (LTC4296_SW_EN_MSK | LTC4296_SW_PSE_READY_MSK |
+				  LTC4296_SET_CLASSIFICATION_MODE_MSK));
+}
+
+/**
+ * @brief Set port to maintain full voltage signature.
+ * @param dev - The device structure
+ * @param port_no - The port number
+ * @return 0 in case of success, negative code otherwise
+ */
+int ltc4296_set_port_mfvs(struct ltc4296_dev *dev, enum ltc4296_port port_no)
+{
+	int ret;
+	uint8_t port_addr = 0;
+	uint16_t val16;
+	int mfvs_threshold, val;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ltc4296_get_port_addr(port_no, LTC_PORT_ADCCFG, &port_addr);
+	if (ret)
+		return ret;
+
+	/* LTC4296-1 Set Port ADC MFVS Threshold Value */
+	val = ltc4296_spoe_sense_resistor[port_no];
+	mfvs_threshold = (625 * val / 10);
+	/* Round of to the nearest integer */
+	val16 = round(mfvs_threshold) / 1000;
```

Comment:
`round` doesn't seem to be defined anywhere. Am I missing something?

Reply:
you're right... however I'm not sure a round is required since `mfvs_threshold` is an integer.